### PR TITLE
docs(functions): document calling functions with parameters 

### PR DESCRIPTION
**Pull request by PaulLVoelker**
## Description
Added description of how to use endpoint calling with parameters and event-based cloud functions.
With that addition, the current possibilities of cloud functions in combination with flutter are completed.
This would have saved me a lot of time if it were present before.
## Related Issues
-

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/functions/usage.mdx [hunk 1]
```diff
@@ -49,6 +49,56 @@ Future<void> getFruit() async {
...
+
+    const uppercase = original.toUpperCase();
+
+    // You must return a Promise when performing asynchronous tasks inside a Functions such as
+    // writing to Firestore.
+    // Setting an 'uppercase' field in Firestore document returns a Promise.
+    return snap.ref.set({ uppercase }, { merge: true });
+  });
+
+```
```
Comment (russellwheatley) [inline docs/functions/usage.mdx:101]: Please remove this, it's not relevant to Flutterfire.

File: docs/functions/usage.mdx [hunk 2]
```diff
@@ -49,6 +49,56 @@ Future<void> getFruit() async {
   List fruit = results.data;  // ["Apple", "Banana", "Cherry", "Date", "Fig", "Grapes"]
 }
 ```
+### Calling Endpoints with Parameters
+Let's have a look at calling Cloud functions with additional Parameters. For example:
```
Comment (russellwheatley) [inline docs/functions/usage.mdx:53]: space above sentence

File: docs/functions/usage.mdx [hunk 3]
```diff
@@ -49,6 +49,56 @@ Future<void> getFruit() async {
...
+
+```dart
+import 'package:cloud_functions/cloud_functions.dart';
+
+Future<void> writeMessage(String message) async {
+    HttpsCallable callable = FirebaseFunctions.instance.httpsCallable('writeMessage');
+    final resp = await callable.call(<String, dynamic>{
+      'text': 'A message send from a client device',
+    });
+    print("result: ${resp.data}");
```
Comment (russellwheatley) [inline docs/functions/usage.mdx:74]:
```suggestion
    print("result: ${resp.data}"); // 'Successfully received: A message send from a client device'
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Thanks for the PR. Please make amendments and sign the CLA. Thanks.

File: docs/functions/usage.mdx (hunk 2 above)
Comment (russellwheatley) [inline docs/functions/usage.mdx:53]:
```suggestion
Let's have a look at calling Cloud functions with additional Parameters:
```

**Review by PaulLVoelker — COMMENTED** [review]
Resolved

**Review by PaulLVoelker — COMMENTED** [review]
Please check again

**Comment by russellwheatley** [comment]
@PaulLVoelker Still some amendments to make. Have you signed the CLA as well?

**Comment by PaulLVoelker** [comment]
I am having trouble signing the CLA agreement. It always shows `Invalid GitHub login.` Do you have any support links?

**Comment by russellwheatley** [comment]
@PaulLVoelker  https://cla.developers.google.com/about

**Comment by PaulLVoelker** [comment]
CLA signed.

File: docs/functions/usage.mdx [hunk 4]
```diff
@@ -50,6 +50,32 @@ Future<void> getFruit() async {
...
+  }
+``
```
Comment (Salakar) [inline docs/functions/usage.mdx:77]: Should be 3 back ticks here

File: docs/functions/usage.mdx [hunk 5]
```diff
@@ -50,6 +50,32 @@ Future<void> getFruit() async {
...
+
+Within Flutter, the function can be called, and the results accessed:
+
+```dart
+import 'package:cloud_functions/cloud_functions.dart';
+
+Future<void> writeMessage(String message) async {
+    HttpsCallable callable = FirebaseFunctions.instance.httpsCallable('writeMessage');
+    final resp = await callable.call(<String, dynamic>{
+      'text': 'A message send from a client device',
```
Comment (russellwheatley) [inline docs/functions/usage.mdx:73]:
```suggestion
      'text': 'A message sent from a client device',
```

**Comment by russellwheatley** [comment]
@PaulLVoelker Yes, there are 3 changes that need to be resolved. Please don't press the resolved button until they are actually resolved. Thanks.

File: docs/functions/usage.mdx [hunk 6]
```diff
@@ -50,6 +50,32 @@ Future<void> getFruit() async {
 }
 ```
 
+### Calling Endpoints with Parameters
```
Comment (Salakar) [inline docs/functions/usage.mdx:53]:
```suggestion
### Calling Endpoints with Parameters

```

**Review by Salakar — APPROVED** [review]

**Comment by Salakar** [comment]
Much thanks for this :)

**Pull request merged**